### PR TITLE
Add sql mode format restrictions to expr in ticdc filter

### PR DESCRIPTION
**Pull request by hongyunyan**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): https://github.com/pingcap/tiflow/pull/10644

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by asddongmen** [comment]
/hold 
Need more test.

**Review by asddongmen — APPROVED** [review]

File: ticdc/ticdc-filter.md [hunk 1]
```diff
@@ -76,12 +76,34 @@ ignore-update-new-value-expr = "gender = 'male' and age > 18" # 过滤掉新值
...
+| set default value | DDL  | |匹配 set default value event    |
+| add primary key | DDL  | |匹配 add primary key event    |
+| drop primary key | DDL  | |匹配 drop primary key event    |
+| rename index | DDL  | |匹配 rename index event    |
+| alter index visibility | DDL  | |匹配 alter index visibility event    |
+| alter ttl info | DDL  | |匹配 alter ttl info event    |
+| alter ttl remove| DDL  | |匹配 alter table remove ttl info event    |
+| multi schema change | DDL  | |匹配在同一条 DDL 语句内对一个表的多个属性进行更改的 DDL event  |
+
+注意: 由于 TiDB 的 DDL 语句是支持同时变更单个表的多个属性操作的，例如 `ALTER TABLE t MODIFY COLUMN a INT, ADD COLUMN b INT, DROP COLUMN c;` 这种这种操作会被定义为 MultiSchemaChange，你如果想过滤掉这种类型的 DDL，需要配置 "multi schema change" 。
```
Comment (lilin90) [inline ticdc/ticdc-filter.md:100]:
```suggestion
> **注意：**
>
> TiDB 的 DDL 语句支持同时变更单个表的多个属性，例如 `ALTER TABLE t MODIFY COLUMN a INT, ADD COLUMN b INT, DROP COLUMN c;` 这种操作会被定义为 MultiSchemaChange。如果想过滤掉这种类型的 DDL，需要配置 "multi schema change"。
```

**Review by lilin90 — APPROVED** [review]
/lgtm

File: ticdc/ticdc-filter.md [hunk 2]
```diff
@@ -57,31 +57,55 @@ ignore-update-new-value-expr = "gender = 'male' and age > 18" # 过滤掉新值
 - `matcher`：该事件过滤器所要匹配的数据库名和表名，其匹配规则和[表库过滤规则](/table-filter.md#表库过滤语法)相一致。
 - `ignore-event`：要过滤的事件类型，它是一个字符串数组，可以配置多个事件类型。目前支持的类型如下表所示:
```
Comment (qiancai) [inline ticdc/ticdc-filter.md:58]:
```suggestion
- `ignore-event`：要过滤掉的事件类型，它是一个字符串数组，可以配置多个事件类型。目前支持的类型如下表所示:
```

File: ticdc/ticdc-filter.md [hunk 3]
```diff
@@ -57,31 +57,55 @@ ignore-update-new-value-expr = "gender = 'male' and age > 18" # 过滤掉新值
...
+    | insert          | DML  | |匹配 insert DML event      |
+    | update          | DML  | |匹配 update DML event      |
+    | delete          | DML  | |匹配 delete DML event      |
+    | create schema   | DDL  | create database |匹配 create database event |
+    | drop schema     | DDL  | drop database  |匹配 drop database event |
+    | create table    | DDL  | |匹配 create table event    |
+    | drop table      | DDL  | |匹配 drop table event      |
+    | rename table    | DDL  | |匹配 rename table event    |
+    | truncate table  | DDL  | |匹配 truncate table event  |
+    | alter table     | DDL  | |匹配 alter table event (包含 alter table 的所有子句和 create/drop index)     |
```
Comment (qiancai) [inline ticdc/ticdc-filter.md:73]:
```suggestion
    | alter table     | DDL  | |匹配 alter table event（包含 alter table 的所有子句和 create/drop index）     |
```

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**